File: Assets/RogueLike/Scripts/CameraController.js
class CameraController extends MonoBehaviour{

	//Focus point of camera
	public var defaultFocus : GameObject;
	
	//Offset to show angle
	public var offsetX : float;
	public var offsetY : float;
	
	//zoom clamping
	public var minDistance : float;
	public var maxDistance : float;
	public var increment : float;
	
	//cached objects
	private var thisTransform : Transform;

	function Start(){
		thisTransform = transform;
	}
	function Update () {
	
		//Update the position to avatar + an offset
		thisTransform.position.x = defaultFocus.transform.position.x + offsetX;
		thisTransform.position.z = defaultFocus.transform.position.z + offsetY;
		
		//perform zoom -orthograpic only
		if (Input.GetAxis("Mouse ScrollWheel") < 0) // back
		{
			camera.main.orthographicSize = Mathf.Max(Camera.main.orthographicSize - increment , minDistance);
			 
		}
		if (Input.GetAxis("Mouse ScrollWheel") > 0) // forward
		{
			camera.main.orthographicSize = Mathf.Min(Camera.main.orthographicSize + increment, maxDistance);
		
		}
	}
	
	public function changeFocus(object : GameObject){
		this.defaultFocus = object;

	}
}

File: Assets/ProceduralCave/Scripts/CameraController.cs
using UnityEngine;
using System.Collections;

public class CameraController : MonoBehaviour
{
	
		//Focus point of camera
		public GameObject defaultFocus;
		private Camera camera;
		public float distance;
		public float angle;
		public float height;

		//zoom clamping
		public float minDistance;
		public float maxDistance;
		public float increment;
	
		//cached objects
		private Transform thisTransform;
		public Transform parentTransform;
	
		void Start ()
		{
				thisTransform = transform;
				parentTransform = transform.parent;
				camera = Camera.main;
				this.transform.parent.transform.position = defaultFocus.transform.position;
				camera.fieldOfView = 5;
				moveCamera ();
		}
	
		private void followPlayer ()
		{
				this.transform.parent.transform.position = defaultFocus.transform.position;
		}

		private void moveCamera ()
		{
				var xp = distance * Mathf.Cos (angle);
				var zp = distance * Mathf.Sin (angle);
				Vector3 newPosition = new Vector3 (xp, height, zp);
				transform.localPosition = newPosition;
				//Make sure the camera is still looking at the player.
				transform.LookAt (defaultFocus.transform.position);
		}
	
		public void spinCamera (float dif)
		{

				if (dif > 0) {
						angle += .05f;
				} else if (dif < 0) {
						angle -= .05f;
				}

				moveCamera ();

		}

		void Update ()
		{

				//base position stays on player
				this.transform.parent.transform.position = defaultFocus.transform.position;

				//perform zoom -orthograpic only
				if (Input.GetAxis ("Mouse ScrollWheel") < 0) { // back
						if (camera.orthographic) {
								camera.orthographicSize = Mathf.Max (Camera.main.orthographicSize - increment, minDistance);
						} else {
								camera.fieldOfView = Mathf.Max (camera.fieldOfView - increment, minDistance);
						}
				}
				if (Input.GetAxis ("Mouse ScrollWheel") > 0) { // forward
						if (camera.orthographic) {
								camera.orthographicSize = Mathf.Min (Camera.main.orthographicSize + increment, maxDistance);
						} else {
								camera.fieldOfView = Mathf.Min (camera.fieldOfView + increment, maxDistance);
						}
				}
		}
	
		public void changeFocus (GameObject o)
		{
				this.defaultFocus = o;
		
		}
}
